crypto/ocsp/ocsp_server.c

```diff
@@ -172,20 +172,37 @@ static int OCSP_basic_sign_ctx(OCSP_BASICRESP *resp, X509 *signer,
   return 1;
 }
 
+const EVP_MD *OCSP_get_default_digest(const EVP_MD *dgst, EVP_PKEY *signer) {
+  if (dgst != NULL) {
+    return dgst;
+  }
```

Comment:
NP: Why have this function take `dgst` as a parameter? It would seem more natural to have the callers conditionally invoke `OCSP_get_default_digest` only when `dgst` is NULL.

Reply:
Agreed, that was the original way I had implemented it. But we would need double null checks (outlined below) in both functions without the abstraction into `OCSP_get_default_digest`. It didn't sit quite well with me since we were already trying to abstract out shared logic, which was why I moved it into the static function.
```
const EVP_MD *init_dgst = dgst;
if (init_dgst != NULL) {
    init_dgst = OCSP_get_default_digest(key);
}
if (init_dgst != NULL) {
  OPENSSL_PUT_ERROR(OCSP, EVP_R_NO_DEFAULT_DIGEST);
   goto err;
}
```